gammapy/spectrum/extract.py

```diff
@@ -218,22 +226,20 @@ def apply_containment_correction(self, obs, bkg):
         else:
             psf = obs.psf.to_energy_dependent_table_psf(offset, angles)
 
-        center_energies = self._on_vector.energy.nodes
-        areascal = []
+        center_energies = self._aeff.energy.nodes
+        containment = []
         for index, energy in enumerate(center_energies):
             try:
-                correction = psf.integral(energy,
-                                          0. * u.deg,
-                                          bkg.on_region.radius)
+                cont_ = psf.integral(energy, 0. * u.deg, bkg.on_region.radius)
             except:
                 msg = 'Containment correction failed for bin {}, energy {}.'
                 log.warning(msg.format(index, energy))
-                correction = 1
+                cont_ = 1
             finally:
-                areascal.append(correction)
+                containment.append(cont_)
 
-        self._on_vector.areascal = areascal
-        self._off_vector.areascal = areascal
+        self.containment = containment
```

Comment:
Wouldn't it be better to make self.containment a numpy array?

Reply:
yes